### PR TITLE
fix: report correct DatabaseStateCode

### PR DESCRIPTION
**Pull request by tustvold**
The original cut of the server status API didn't include the error states as discrete states, this was supposed to be changed in #2203 but I copy-pasta'ed the conversion logic and didn't notice that I hadn't updated the tests :facepalm: 

This PR fixes this and also adds the enumerations for the states added since then.

I also opted to remove the manually defined enumeration in data_types as its only purpose seemed to be to define an impl, which can just be defined in generated_types and some duplication avoided.

### CODE REVIEW

File: generated_types/protos/influxdata/iox/management/v1/service.proto [hunk 1]
```diff
@@ -386,6 +389,9 @@ message DatabaseStatus {
 
     // Error during replay
     DATABASE_STATE_REPLAY_ERROR = 7;
+
+    // Error encountered identifying active generation
+    DATABASE_STATE_DATABASE_OBJECT_STORE_LOOKUP_ERROR = 9;
```
Comment (tustvold) [inline generated_types/protos/influxdata/iox/management/v1/service.proto:394]: This enumeration name makes me irrationally sad, but I couldn't think of a better name. We can always change it later

File: generated_types/protos/influxdata/iox/management/v1/service.proto [hunk 2]
```diff
@@ -177,16 +177,16 @@ message DeleteDatabaseResponse {}
...
 // This resource represents a deleted database.
 message DeletedDatabase {
-	// The generation ID of the deleted database.
+  // The generation ID of the deleted database.
   uint64 generation_id = 1;
 
-	// The UTC datetime at which this database was deleted.
-	google.protobuf.Timestamp deleted_at = 2;
+  // The UTC datetime at which this database was deleted.
+  google.protobuf.Timestamp deleted_at = 2;
```
Comment (tustvold) [inline generated_types/protos/influxdata/iox/management/v1/service.proto:189]: Perhaps we need a format check for protobuf :thinking:

Comment (alamb) [reply generated_types/protos/influxdata/iox/management/v1/service.proto:189]: maybe the `buf` thing we use could be pressed into service

File: generated_types/protos/influxdata/iox/management/v1/service.proto (hunk 1 above)
Comment (alamb) [reply generated_types/protos/influxdata/iox/management/v1/service.proto:394]: its A DATABASE! A DATABASE!

**Review by alamb — APPROVED** [review]

**Review by carols10cents — APPROVED** [review]
When I started looking into this, I started adding some cases to ensure we could get some of the other failing database states-- so normally I *would* ask you to add that, but I have it halfway done anyway so I'll add it in a new PR :)

**Comment by carols10cents** [comment]
Connects to #2539

**Pull request merged**